qcengine/programs/dftd3.py

```diff
@@ -62,326 +61,224 @@ def get_version(self) -> str:
-                       modulerec.get('infiles'),
-                       modulerec.pop('outfiles'),
-                       **{
-        'scratch_messy': False,
-        'environment': modulerec.get('env'),
-        'blocking_files': modulerec.get('blocking_files')
-    })
-    modulerec.update(dexe)  # updates modulerec
-
-    if verbose > 3:
-        print(f'[3] {module_label.upper()}REC POST-SUBPROCESS (m@io) <<<')
-        pp.pprint(modulerec)
-        print('>>>\n')
-
-    harvest(jobrec, modulerec)  # updates jobrec
-
-    if verbose > 1:
-        print(f'[4] {module_label.upper()} JOBREC POST-HARVEST (j@io) <<<')
-        pp.pprint(jobrec)
-        print('>>>')
 
-    return jobrec
-
-
-def dftd3_plant(jobrec):
-    """Transform the QC input specifications `jobrec` into the command
-    and files for DFTD3: jobrec@i -> dftd3rec@i.
-
-    Parameters
-    ----------
-    jobrec : dict
-        Nested dictionary with input specifications for DFTD3 in generic
-        QC terms.
-
-    Returns
-    -------
-    dftd3rec : dict
-        Nested dictionary with input specification for DFTD3 in
-        program-specific commands and files.
-
-    """
-    # temp until actual options object
-    jobrec['extras']['info'] = empirical_dispersion_resources.from_arrays(
-        name_hint=jobrec['model']['method'],
-        level_hint=jobrec['keywords'].get('level_hint', None),
-        param_tweaks=jobrec['keywords'].get('params_tweaks', None),
-        dashcoeff_supplement=jobrec['keywords'].get('dashcoeff_supplement', None))
-
-    # this is what the dftd3 program needs, not what the job needs
-    # * form dftd3_parameters string that governs dispersion calc
-    # * form dftd3_geometry string that supplies geometry to dispersion calc
-    # * form command and arguments
-
-    modulerec = {}
-    modulerec['infiles'] = {}
-    modulerec['infiles']['.dftd3par.local'] = dftd3_coeff_formatter(jobrec['extras']['info']['dashlevel'],
-                                                                   jobrec['extras']['info']['dashparams'])
-
-    # Need 'real' field later and that's only guaranteed for molrec
-    molrec = qcel.molparse.from_schema(jobrec['molecule'])
-    modulerec['infiles']['dftd3_geometry.xyz'] = qcel.molparse.to_string(
-        molrec, dtype='xyz', units='Angstrom', ghost_format='')
-
-    modulerec['outfiles'] = [
-        'dftd3_gradient',
-        'dftd3_abc_gradient',
-    ]
-    modulerec['env'] = {
-            'HOME': os.environ.get('HOME'),
-            'PATH': os.pathsep.join([os.path.abspath(x) for x in os.environ.get('PSIPATH', '').split(os.pathsep) if x != '']) + \
-                    os.pathsep + os.environ.get('PATH'),
-            'LD_LIBRARY_PATH': os.environ.get('LD_LIBRARY_PATH'),
+        return {
+            "command": command,
+            "infiles": infiles,
+            "outfiles": ['dftd3_gradient', 'dftd3_abc_gradient'],
+            "scratch_location": config.scratch_directory,
```

Comment:
We should standardize location/directory one way or the other across the whole code base. I vote a bit for `scratch_directory` as this keyword is found in the large QCA ecosystem.

Reply:
If it meant the place directly where the calc happens, I would agree. But `scratch_location` turns into `parent` https://github.com/MolSSI/QCEngine/blob/master/qcengine/util.py#L418 or "scratch directory directory." Is that what the generic QCA `scratch_directory` means?

Reply:
I mean from a QCA perspective it doesn't matter. It is the location to spawn tasks, if a further tmpdir is built it is quite irrelevant as the tmpdirs are always cleaned. There could be additional complications that I am not considering however.

Reply:
I guess I use “scratch directory” two ways, too. The enclosing loc like /scratch/hohndoe/ or the qc run loc like /tmp/water.4444/. Along that like, I don’t really care directory or location. Can change.